### PR TITLE
[CDAP-16668] Dataproc enhancements to support autoscale (experimental…

### PR DESCRIPTION
**Pull request by CuriousVini**
… feature)

cherry-pick - https://github.com/cdapio/cdap/pull/12129

### CODE REVIEW

File: cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java [hunk 1]
```diff
@@ -436,6 +437,18 @@ public ClusterOperationMetadata createCluster(String name, String imageVersion,
             .build());
       }
 
+      //Set Auto Scaling Policy
+      if (conf.isAutoPolicyEnabled()) {
```
Comment (chtyim) [inline cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java]: Why we need two separate config? Just the policyURI is good enough, right? If it is not null nor empty, then we'll apply the policy

Comment (CuriousVini) [reply cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java]: yea we dont need it

**Comment by CuriousVini** [comment]
https://builds.cask.co/browse/CDAP-RUT1839-1

File: cdap-runtime-ext-dataproc/src/main/resources/gcp-dataproc.json [hunk 2]
```diff
@@ -384,6 +384,20 @@
...
       ]
+    },
+    {
+      "label": "Experimental Features (BETA)",
+      "properties": [
+        {
+          "widget-type": "textbox",
+          "label": "Autoscaling Policy",
+          "name": "autoScalingPolicy",
+          "description": "Specify the Autoscaling Policy id(name) or the resource URI",
```
Comment (chtyim) [inline cdap-runtime-ext-dataproc/src/main/resources/gcp-dataproc.json]: Should there be a space between `id` and `(name)` ?

**Review by chtyim — APPROVED** [review]
Just one minor text spacing

**Pull request merged**